### PR TITLE
refactor: make some header only

### PR DESCRIPTION
**Pull request by wineee**

### CODE REVIEW

File: src/types/qwdrm.h [hunk 1]
```diff
@@ -4,45 +4,25 @@
...
-    }
-
-    static QWDrm *get(wlr_drm *handle);
-    static QWDrm *from(wlr_drm *handle);
-    static QWDrm *create(QWDisplay *display, QWRenderer *render);
-
-private:
-    QWDrm(wlr_drm *handle, bool isOwner);
-    ~QWDrm() = default;
+    QW_FUNC_MEMBER(drm, create)
```
Comment (zccrs) [inline src/types/qwdrm.h:19]: should static?

File: src/types/qwdrm.h [hunk 2]
```diff
@@ -4,45 +4,25 @@
...
+class QW_CLASS_REINTERPRET_CAST(drm_buffer) {
 public:
-    QWDrmBuffer() = delete;
-    ~QWDrmBuffer() = delete;
-
-    wlr_drm_buffer *handle() const;
-
-    static QWDrmBuffer *from(wlr_drm_buffer *handle);
-    static QWDrmBuffer *from(wl_resource *resource);
+    QW_FUNC_MEMBER(drm_buffer, try_from_resource)
```
Comment (zccrs) [inline src/types/qwdrm.h:25]: static?

**Review by zccrs — CHANGES_REQUESTED** [review]

File: src/types/qwdrm.h (hunk 1 above)
Comment (wineee) [reply src/types/qwdrm.h]: Yes

File: src/types/qwdamagering.h [hunk 3]
```diff
@@ -3,37 +3,24 @@
...
-    bool add(pixman_region32_t *damage);
-    bool addBox(const QRect &box);
-    void addWhole();
-    void rotate();
-    void getBufferDamage(int bufferAge, pixman_region32_t *damage) const;
-    void setBounds(const QSize &bounds);
-
-private:
-    std::unique_ptr<wlr_damage_ring> m_handle;
+    QW_FUNC_MEMBER(damage_ring, init)
```
Comment (zccrs) [inline src/types/qwdamagering.h:16]: auto init in constructor

File: src/types/qwdamagering.h [hunk 4]
```diff
@@ -3,37 +3,24 @@
...
+    QW_FUNC_MEMBER(damage_ring, finish)
```
Comment (zccrs) [inline src/types/qwdamagering.h:17]: auto finish in destructor

**Review by zccrs — CHANGES_REQUESTED** [review]

**Review by zccrs — APPROVED** [review]

**Pull request merged**